lib/postgrex/notifications.ex

```diff
@@ -78,18 +78,43 @@ defmodule Postgrex.Notifications do
             listener_channels: %{},
             auto_reconnect: false,
             reconnect_backoff: 500,
-            connected: false
+            connected: false,
+            state: nil
 
   ## PUBLIC API ##
 
+  @type query :: binary()
   @type server :: GenServer.server()
+  @type state :: term
+
+  @doc """
+  Callback for process initialization.
+
+  This is called once and before the Postgrex notification connection is established.
+  """
+  @callback init(term) :: {:ok, state}
+
+  @doc """
+  Callback for `call/3`.
+  """
+  @callback handle_call(term, GenServer.from(), state) ::
+              {:noreply, state} | {:reply, term, state} | {:query, query, state}
+
+  @doc """
+  Callback for processing or relaying query results.
+  """
+  @callback handle_result(Result.t(), state) :: {:noreply, state}
 
   @doc false
   def child_spec(opts) do
-    %{
-      id: __MODULE__,
-      start: {__MODULE__, :start_link, [opts]}
-    }
+    opts =
```

Comment:
Perhaps this match can be done in the function head or with `case`? Don't see the point for `if match?`

Reply:
Agreed, it is pretty silly. This is only a temporary change to get the tests working anyhow. Soon there will be two distinct modules with their own `child_spec/1`.